### PR TITLE
Feature/feature crud health pet data

### PR DESCRIPTION
**Pull request by oricat8**

### CODE REVIEW

File: src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+    @BeforeAll
+    public static void setUp() {
+        weightList = new ArrayList<>();
+        weightEntity = new WeightEntity();
+        date = "2019-10-22T00:47:00";
+        date2 = "2021-10-22T00:47:00";
+        owner = "PericoDeLosPalotes";
+        petName = "TupoJohn";
+        field = "weightValue";
+        value = 9.0 / 2.0;
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java:94]: Codacy Issue found: ['9.0' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 7
           

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/error/DatabaseAccessException.java  75
+ src/main/java/org/pesmypetcare/webservice/firebaseservice/FirebaseFactory.java  20
         

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/error/ErrorBody.java  -75
- src/main/java/org/pesmypetcare/webservice/error/FirebaseExceptionHandler.java  -92
- src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  -38
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/service/WeightServiceImpl.java  1
- src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java  1
- src/test/java/org/pesmypetcare/webservice/service/WeightServiceTest.java  1
- src/test/java/org/pesmypetcare/webservice/controller/petmanager/PetWeightRestController.java  1
- src/main/java/org/pesmypetcare/webservice/controller/petmanager/WeightRestController.java  1
- src/main/java/org/pesmypetcare/webservice/dao/WeightDaoImpl.java  5
         

Clones added
============
- src/main/java/org/pesmypetcare/webservice/service/WeightService.java  1
- src/main/java/org/pesmypetcare/webservice/dao/WeightDao.java  1
- src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java  39
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130&bid=17540865)

File: src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+            given(petsRef.document(anyString())).willReturn(petRef);
+            given(petRef.collection(anyString())).willReturn(weightsRef);
+            given(weightsRef.get()).willReturn(futureQuery);
+            willThrow(ExecutionException.class).given(futureQuery).get();
+
+            weightDao.getAllWeightsBetween(owner, petName, date, date2);
+        }, EXCECUTION_EXC_MSG);
+    }
+
+    @Test
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java:366]: Codacy Issue found: [Method length is 17 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

File: src/main/java/org/pesmypetcare/webservice/dao/WeightDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+     */
+    private void getWeightsBetweenDatesFromDatabase(String initialDate, String finalDate,
+                                                    CollectionReference weightsRef, List<Map<String,
+        Object>> externalList) throws InterruptedException, ExecutionException {
+        ApiFuture<QuerySnapshot> future = weightsRef.get();
+        List<QueryDocumentSnapshot> weightDocuments = future.get().getDocuments();
+        for (QueryDocumentSnapshot weightDocument : weightDocuments) {
+            String date = weightDocument.getId();
+            if (initialDate.compareTo(date) < 0 && finalDate.compareTo(date) > 0) {
+                Map<String, Object> internalList = new HashMap<>();
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/WeightDaoImpl.java:159]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

File: src/test/java/org/pesmypetcare/webservice/service/WeightServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    @BeforeAll
+    public static void setUp() {
+        weightList = new ArrayList<>();
+        weightEntity = new WeightEntity();
+        date = "2020-02-13T10:30:00";
+        date2 = "2021-02-13T10:30:00";
+        owner = "TupoJohn";
+        petName = "Perico";
+        value = 9.0 / 2.0;
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/WeightServiceTest.java:50]: Codacy Issue found: ['9.0' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

File: src/test/java/org/pesmypetcare/webservice/service/WeightServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class WeightServiceTest {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/service/WeightServiceTest.java:26]: Codacy Issue found: [This class has too many methods, consider refactoring it.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

File: src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java (hunk 2 above)
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java:366]: Codacy Issue found: [NCSS for this method is 16 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

File: src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class WeightDaoTest {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/WeightDaoTest.java:37]: Codacy Issue found: [This class has too many methods, consider refactoring it.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/error/DatabaseAccessException.java  75
+ src/main/java/org/pesmypetcare/webservice/firebaseservice/FirebaseFactory.java  20
         

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  -38
- src/main/java/org/pesmypetcare/webservice/error/ErrorBody.java  -75
- src/main/java/org/pesmypetcare/webservice/error/FirebaseExceptionHandler.java  -92
         

Complexity increasing per file
==============================
- src/main/java/org/pesmypetcare/webservice/controller/petmanager/WeightRestController.java  1
- src/test/java/org/pesmypetcare/webservice/controller/petmanager/PetWeightRestController.java  1
- src/main/java/org/pesmypetcare/webservice/service/WeightServiceImpl.java  1
         

Clones added
============
- src/main/java/org/pesmypetcare/webservice/dao/WeightDao.java  1
- src/main/java/org/pesmypetcare/webservice/service/WeightService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5354130&bid=17540865)

**Review by marclll — CHANGES_REQUESTED** [review]
No es pot fer

**Review by marclll — APPROVED** [review]

**Review by marclll — CHANGES_REQUESTED** [review]
Si fariem aquest pull request eliminariem el projecte

**Pull request closed without merging**